src/main/java/com/mixpanel/android/viewcrawler/ViewCrawler.java

```diff
@@ -221,27 +226,14 @@ private void uninstallConnectionSensor(final Activity activity) {
         }
 
         private boolean isInEmulator() {
-            if (!Build.HARDWARE.equals("goldfish")) {
-                return false;
-            }
-
-            if (!Build.BRAND.startsWith("generic")) {
-                return false;
-            }
-
-            if (!Build.DEVICE.startsWith("generic")) {
-                return false;
-            }
-
-            if (!Build.PRODUCT.contains("sdk")) {
-                return false;
-            }
-
-            if (!Build.MODEL.toLowerCase(Locale.US).contains("sdk")) {
-                return false;
-            }
-
-            return true;
+            return Build.FINGERPRINT.startsWith("generic")
+                    || Build.FINGERPRINT.startsWith("unknown")
```

Comment:
Want to use your example instead of mine? https://github.com/gingo/android-emulator-detector